src/screens/shared/farmSearch/FarmSearch.js

```diff
@@ -12,24 +12,50 @@ class FarmSearch extends React.PureComponent {
   constructor(props) {
     super(props);
     this.state = {
-      farms: []
+      farms: [],
+      search: null,
+      filteredFarms: []
     };
   }
 
   async componentDidMount() {
     const farms = await getAllFarmsForFarmSearch();
-    this.setState({ farms });
+    this.setState({ farms, filteredFarms: farms });
   }
 
-  render() {
+  searchAndFilter = e => {
     const { farms } = this.state;
+    const keyword = e.target.value;
+    const filteredFarms = farms.filter(
+      farm =>
+        (farm.contactFirstName &&
+          farm.contactFirstName
+            .toLowerCase()
+            .includes(keyword.toLowerCase())) ||
+        (farm.farmName &&
+          farm.farmName.toLowerCase().includes(keyword.toLowerCase())) ||
+        (`${farm.contactFirstName} ${farm.contactLastName}` &&
```

Comment:
this will always evaluate to true, because "null null" is a true value

perhaps just check `farm.contactFirstName && farm.contactLastName`

Reply:
hmm oh i see thank you for clarifying!